fire_atlas_editor/lib/services/storage/web.dart

```diff
@@ -0,0 +1,107 @@
+import 'dart:async';
+import 'dart:html';
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:fire_atlas_editor/services/storage/storage.dart';
+import 'package:fire_atlas_editor/store/store.dart';
+import 'package:flame_fire_atlas/flame_fire_atlas.dart';
+
+class FireAtlasStorage extends FireAtlasStorageApi {
+  static final Storage _localStorage = window.localStorage;
+
+  Future<LoadedProjectEntry> loadProject(String path) {
+    final value = _localStorage[path];
+    if (value == null) {
+      throw 'Unknow project with id $path';
+    }
+
+    final entry = LoadedProjectEntry(
+      path,
+      _readBase64Project(value),
+    );
+
+    return Future.value(entry);
+  }
+
+  Future<void> saveProject(LoadedProjectEntry entry) async {
+    final data = entry.project.serialize();
+
+    if (entry.path == null) {
+      entry.path = 'ATLAS_${entry.project.id}';
+    }
+
+    _localStorage[entry.path!] = base64Encode(data);
+  }
+
+  Future<List<LastProjectEntry>> lastUsedProjects() async {
+    return _localStorage.entries
+        .map((e) => e.key)
+        .where((k) => k.startsWith('ATLAS_'))
+        .map((k) {
+      final name = k.replaceFirst('ATLAS_', '');
+      return LastProjectEntry(k, name);
+    }).toList();
```

Comment:
this formatting is a bit odd...

Reply:
That is the fault of `flutter format`, not much to do I think